src/redux/actions/sync.js

```diff
@@ -91,42 +103,34 @@ export function doGetSync(password = '', shouldSetDefaultAccount) {
           data.syncHash = syncHash;
           data.syncData = response.data;
           data.hasSyncedWallet = true;
-          if (response.changed || shouldSetDefaultAccount) {
+          if (response.changed) {
             return Lbry.sync_apply({ password, data: response.data }).then(
               ({ hash: walletHash, data: walletData }) => {
                 dispatch({ type: ACTIONS.GET_SYNC_COMPLETED, data });
 
-                if (walletHash !== syncHash || shouldSetDefaultAccount) {
+                if (walletHash !== syncHash) {
                   // different local hash, need to synchronise
                   dispatch(doSetSync(syncHash, walletHash, walletData));
-
-                  if (shouldSetDefaultAccount) {
-                    dispatch(
-                      doSetDefaultAccount(() => {
-                        Lbry.status().then(status => {
-                          if (status.wallet.is_locked) {
-                            Lbry.account_unlock({ password });
-                          }
-                          dispatch(doFetchChannelListMine());
-                        });
-                      })
-                    );
-                  }
+                  handleCallback();
                 }
               }
             );
           }
 
           dispatch({ type: ACTIONS.GET_SYNC_COMPLETED, data });
+          handleCallback();
```

Comment:
This should probably be in an `else` block unless we want the callback called twice in cases where the first condition is true.

Reply:
Good catch.